sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/literals.scala

```diff
@@ -298,11 +298,21 @@ case class Literal (value: Any, dataType: DataType) extends LeafExpression {
   override def foldable: Boolean = true
   override def nullable: Boolean = value == null
 
+  private val timeZoneId = DateTimeUtils.getZoneId(SQLConf.get.sessionLocalTimeZone)
+
   override def toString: String = value match {
```

Comment:
Does it make sense to have universal conversion of any literal to string via applying of the `Cast` expression to `this` literal?

Reply:
Sorry, I don't understand your question.

Reply:
do you mean this should be
```
override def toString: String = Cast(this, StringType, timeZoneId).eval().toString
```

Reply:
+1 on the @MaxGekk idea.

Reply:
This will change the current behaviour:
```
Literal.create(Array[Byte](65, 66, 67), BinaryType).toString // before: 0x414243, after: ABC
Literal.create(Map("a" -> 1)).toString // before: map(keys: [a], values: [1]), after: {a -> 1}
```
cc @dongjoon-hyun

Reply:
Thank you for pinging me, @wangyum .